onnx_trt_backend.cpp

```diff
@@ -0,0 +1,771 @@
+  }
+
+#define SET_UINT64(x)                                                          \
+  {                                                                            \
+    if (*infoValueSize < sizeof(uint64_t)) {                                   \
+      return ONNXIFI_STATUS_INVALID_POINTER;                                   \
+    }                                                                          \
+    *(uint64_t *)(infoValue) = x;                                              \
+    *infoValueSize = sizeof(uint64_t);                                         \
+  }
+    switch(ONNXIFI_BACKEND_NAME) {
+    case ONNXIFI_BACKEND_NAME:
+    SET_STRING("TensorRT");
+    break;
+    case ONNXIFI_BACKEND_VENDOR:
+    SET_STRING("Nvidia");
+    break;
+    case ONNXIFI_BACKEND_VERSION:
+    SET_STRING("1.0.0");
+    break;
+    case ONNXIFI_BACKEND_EXTENSIONS:
+    *infoValueSize = 0;
+    break;
+    case ONNXIFI_BACKEND_DEVICE:
+    SET_STRING("gpu");
+    break;
+    case ONNXIFI_BACKEND_DEVICE_TYPE:
+    SET_UINT64(ONNXIFI_DEVICE_TYPE_GPU);
+    break;
+    case ONNXIFI_BACKEND_CAPABILITIES:
+    SET_UINT64(0UL);
+    break;
+    case ONNXIFI_BACKEND_INIT_PROPERTIES:
+     SET_UINT64(0UL);
+     break;
+    case ONNXIFI_BACKEND_MEMORY_TYPES:
+     SET_UINT64(ONNXIFI_MEMORY_TYPE_CUDA_BUFFER);
+     break;
+    case ONNXIFI_BACKEND_MEMORY_SIZE:
+     {
+      size_t free, total;
+      if (cudaMemGetInfo(&free, &total) != cudaSuccess) {
+        return ONNXIFI_STATUS_BACKEND_UNAVAILABLE;
+      }
+      SET_UINT64(uint64_t(total));
+      break;
+     }
+    // TODO: Dummy numbers below
+    case ONNXIFI_BACKEND_MAX_GRAPH_SIZE:
+     SET_UINT64(1000000UL);
+     break;
+    case ONNXIFI_BACKEND_MAX_GRAPH_COUNT:
+     SET_UINT64(1UL);
+     break;
+    case ONNXIFI_BACKEND_MACS_FP32:
+     SET_UINT64(0UL);
+     break;
+    case ONNXIFI_BACKEND_MACS_FP16:
+     SET_UINT64(0UL);
+     break;
+    case ONNXIFI_BACKEND_MEMORY_BANDWIDTH:
+     SET_UINT64(0UL);
+     break;
+    case ONNXIFI_BACKEND_CPU_MEMORY_READ_BANDWIDTH:
+     SET_UINT64(0UL);
+     break;
+    case ONNXIFI_BACKEND_CPU_MEMORY_WRITE_BANDWIDTH:
+     SET_UINT64(0UL);
+     break;
+    default:
+     return ONNXIFI_STATUS_UNSUPPORTED_PARAMETER;
+    }
+    return ONNXIFI_STATUS_SUCCESS;
+#undef SET_STRING
+#undef SET_UINT64
+  });
+}
+
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
+    onnxGetBackendCompatibility)(onnxBackendID backendID, size_t onnxModelSize,
+                                 const void *onnxModel) {
+  return OnnxifiTryCatch([&] {
+    if (!onnxModel) {
+      return ONNXIFI_STATUS_INVALID_POINTER;
+    }
+    if (onnxModelSize == 0) {
+      return ONNXIFI_STATUS_INVALID_SIZE;
+    }
+
+    // NB: not ideal case. We CHECK model by actually trying to run the
+    // conversion. However, this might be the case for other vendors
+    auto *backend_id = reinterpret_cast<OnnxTensorRTBackendID *>(backendID);
```

Comment:
Conversion always succeeds because it is just a pointer cast. If you want to be safe, record `magic` field in ONNXIFI-TRT objects, e.g. like here: https://github.com/Maratyszcza/onnx/blob/onnxifi-wrapper/onnx/onnxifi_wrapper.c#L43-L70 ("magic" values, of course, must be different)